### ISSUE TITLE
twisted/protocols/stateful.py uses StringIO incorrectly

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2355]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2355|
|Type|defect|
|Created|2007-01-05 15:20:04Z|
|Branch|https://github.com/twisted/twisted/tree/diaf-stringio-2355|

The module tries to import cStringIO and falls back to StringIO.  It then uses the ```reset``` method, which is only defined by the cStringIO version.

<details><summary>Searchable metadata</summary>

```
trac-id__2355 2355
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__branches_diaf_stringio_2355 branches/diaf-stringio-2355
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1168010404000000 1168010404000000
changetime__1211317688000000 1211317688000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="pahan's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|pahan commented|
|-|-|

(In [23645]) Branching to diaf-string-2355

**Comment by twisted-trac** [comment]
|<img alt="pahan's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|pahan commented|
|-|-|

(In [23648]) Branching to 'diaf-stringio-2355'

**Comment by twisted-trac** [comment]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|PenguinOfDoom commented|
|-|-|

Who uses StringIO these days, anyway :/

**Comment by twisted-trac** [comment]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|PenguinOfDoom removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to PenguinOfDoom|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|<img alt="pahan's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|pahan set status to `closed`|
|-|-|

(In [23669]) Merge diaf-stringio-2355

Author: PenguinOfDoom
Reviewer: therve
Fixes: [#2355](https://github.com/twisted/twisted/issues/2355)

Do not use StringIO.reset()

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
